lib/rummager/app.rb

```diff
@@ -73,7 +73,17 @@ def json_only
     halt(400, env['sinatra.error'].message)
   end
 
-  # Return results for the GOV.UK site search
+  error MetasearchIndex::Deleter::V2::UnknownError, MetasearchIndex::Deleter::V2::MissingArgument,
+    MetasearchIndex::Inserter::V2::UnknownError, MetasearchIndex::Inserter::V2::MissingArgument do
+    halt(500, env['sinatra.error'].message)
```

Comment:
This should be a 400 error as it's only thrown if the client has passed us an invalid message

Reply:
changed